### DISCUSSION TITLE
Incorrect comment?

### DISCUSSION THREAD

**Discussion by AriaAdibi** [google/auto#1291] [Q&A] [answered]
https://github.com/google/auto/blob/00fe738a9d92a5b962c6b01c99a14b848435d882/common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java#L522

I have analyzed the code and I believe the code does not behave as this comment suggests. I believe the deferral policy of (any kind of) package is the same as that of `TypeElement`s. The only peculiar thing about packages is that to check for well-informedness only their annotations are superficially validated and nothing else. Moreover, there is no mention of `package-info.java` in the code. Am I wrong?

**Comment by eamonnmcmanus** [comment, marked as answer]
As you say, validating a package consists only of validating the annotations on it. The point about `package-info.java` is that that is the mechanism for annotating a package:

```
// com/example/foo/package-info.java

@SomeAnnotation
@SomeOtherAnnotation
package com.example.foo;
```

Suppose package `com.example.foo` was one of the elements returned by [`RoundEnvironment.getElementsAnnotatedWith`](https://docs.oracle.com/en/java/javase/17/docs/api/java.compiler/javax/annotation/processing/RoundEnvironment.html#getElementsAnnotatedWith(javax.lang.model.element.TypeElement)), and suppose the class `SomeOtherAnnotation` doesn't exist (because an annotation processor will create it later). Then we will defer an `ElementName` representing package `com.example.foo` to a later processing round.

I don't think the comment is _wrong_ per se, but it could possibly be worded better. Feel free to send a PR with a wording that you would have found easier to follow.
